src/test/scala/com/cognite/spark/datasource/TimeSeriesUpsertTest.scala

```diff
@@ -91,6 +92,191 @@ class TimeSeriesUpsertTest extends FlatSpec with SparkTest {
     assert(initialDescriptionsAfterUpsert.count == 0)
   }
 
+  it should "support inserts, updates and upserts in savemode" taggedAs WriteTest in {
+    val insertDescription = "spark-insert-test"
+    val updateDescription = "spark-update-test"
+    val upsertDescription = "spark-upsert-test"
+    val saveModeUnit = "spark-savemode-test"
+
+    // Clean up any old test data
+    val namesDf =
+      spark.sql(s"""select name from sourceTimeSeries where unit = '$saveModeUnit'""")
+    val namesList = namesDf.rdd.map(r => r.getAs[String](0)).collect().toList
+    cleanupTimeSeries(namesList)
+
+    // Insert new time series test data
+      spark.sql(s"""
+              |select '$insertDescription' as description,
+              |isString,
+              |concat('TEST_', name) as name,
+              |metadata,
+              |'$saveModeUnit' as unit,
+              |NULL as assetId,
+              |isStep,
+              |cast(array() as array<long>) as securityCategories,
+              |id+2 as id,
```

Comment:
Isn't id ignored anyway on inserts? The user can't set this. Clearer to use `null` here as well, I think?

Reply:
Good catch, turns out id could be deleted from the PostTimeSeriesItem class as well.